include/common/errcode.h

```diff
@@ -32,13 +32,71 @@
 
 namespace WasmEdge {
 
-static inline WasmPhase getErrCodePhase(ErrCode Code) {
-  return static_cast<WasmPhase>((static_cast<uint8_t>(Code) & 0xF0) >> 5);
+class ErrCode {
+public:
+  constexpr ErrCategory getCategory() const noexcept {
+    return static_cast<ErrCategory>(static_cast<uint8_t>(Inner.Num >> 24));
```

Comment:
Considering the `endianess` issue, we should use `array (Your code in C)` or `shift (Your code in C++)` in both C and C++ implementations.